custom/rec/collections.py

```diff
@@ -0,0 +1,17 @@
+def dicts_or(*dicts):
+    """
+    Returns a dictionary of key-values from d1 | d2 | d3 ...
+
+    >>> foo = {1: 'one', 2: 'two'}
+    >>> bar = {2: 'deux', 3: 'trois'}
+    >>> baz = {3: 'drie', 4: 'vier'}
+    >>> dicts_or(foo, bar, baz)
+    {1: 'one', 2: 'two', 3: 'trois', 4: 'vier'}
+
+    """
+    assert len(dicts), "No operands"
+    if len(dicts) == 1:
+        return dicts[0]
+    d1 = dicts[0]
+    dn = dicts_or(*dicts[1:])
+    return {k: d1[k] if k in d1 else dn[k] for k in d1.keys() | dn.keys()}
```

Comment:
Is this any different than:

```python
ret = {}
for d in reverse(dicts):
    ret.update(d)
return ret
```

I reversed ret because `update` appears to have the opposite behavior of what you were going for.

However looking at your only usage of this in the next commit, I don't think that order matters

Reply:
this looks fun - I'll throw my hat in the ring with
```python
 def dicts_or(*dicts): 
    return dict(chain(*(d.items() for d in dicts)))
```
or the slightly less fancy
```python
 def dicts_or(*dicts): 
    return {k: v for d in dicts for k, v in d.items()} 
```
possibly also with a `reversed` in there if necessary as @emord described

Reply:
Oh yeah! Those are much better! Nice!